spock-specs/src/test/groovy/org/spockframework/docs/utilities/concurrent/BlockingVariablesDocSpec.groovy

```diff
@@ -0,0 +1,48 @@
+package org.spockframework.docs.utilities.concurrent
+
+import spock.lang.Specification
+import spock.util.concurrent.BlockingVariable
+import spock.util.concurrent.BlockingVariables
+
+class BlockingVariablesDocSpec extends Specification {
+
+  // tag::blocking-variables-spec[]
+  def "single variable is read before it is written"() {
+    def list = new BlockingVariable<List<Integer>>()          // <1>
+
+    when:
+    Thread.start {
+      Thread.sleep(250)          // <2>
+      println "calling set"
+      list.set([1, 2, 3])
+    }
+
+    then:
+    println "calling get, blocking"
+    list.get() == [1, 2, 3]          // <3>
+  }
+
+  def "example of multiple variables"() {
+    def vars = new BlockingVariables(2.0)          // <4>
+
+    when:
+    Thread.start {
+      Thread.sleep(500)          // <5>
```

Comment:
I made some tests, i.e. repeating the specs ~50 times, and it seemed that (on my machine) the sleeps could go as low as about 10ms before the order of execution for some of the statements would be changing. So I chose 25ms and 50ms respectively in the commit I just made.
Mind that there are also some timeouts in the *actual* tests, i.e. in the `spock.util.concurrent` package, which is where I got the numbers from in the first place. I considered lowering them also, but didn't want to without your approval. Let me know, and I will change those numbers also.

Reply:
Unfortunately, the CI servers are slower than most PCs so that is why there is a bit of a margin in the `BlockingVariable(s)Spec`. It could probably be lowered to 250 without making the tests too flaky on ci.

However, if the time is not really critical to the test it should be as low possible. In these tests it is just used to visualize a delay, the correctness of `BlockingVariable` is already tested in the other specs.